pulser-core/pulser/json/abstract_repr/deserializer.py

```diff
@@ -443,20 +454,33 @@ def deserialize_abstract_sequence(obj_str: str) -> Sequence:
         # This is kept for backwards compatibility
         seq.config_slm_mask(obj["slm_mask_targets"])
 
-    # Detuning Map configuration
+    to_config_det_map: dict[str, dict] = dict()
     if "dmm_channels" in obj:
-        for dmm_id, ser_det_map in obj["dmm_channels"]:
-            trap_coords = []
-            weights = []
-            for trap in ser_det_map["traps"]:
-                trap_coords.append((trap["x"], trap["y"]))
-                weights.append(trap["weight"])
-            det_map = DetuningMap(
-                trap_coordinates=trap_coords,
-                weights=weights,
-                slug=ser_det_map.get("slug"),
+        # Make a dictionnary with all the configured dmm channels
+        for dmm_name, ser_det_map in obj["dmm_channels"]:
+            splitted_dmm_name = dmm_name.split("_")  # dmm_id_num
+            dmm_id = "_".join(splitted_dmm_name[0:2])
+            call_num = (
+                0 if len(splitted_dmm_name) <= 2 else int(splitted_dmm_name[2])
             )
-            seq.config_detuning_map(detuning_map=det_map, dmm_id=dmm_id)
+            if dmm_id not in to_config_det_map:
+                to_config_det_map[dmm_id] = {}
+            to_config_det_map[dmm_id][call_num] = ser_det_map
+
+        for dmm_id, ser_det_maps in to_config_det_map.items():
+            slm_mask_dmm = None
+            call_nums = list(ser_det_maps.keys())
+            # If an SLM Mask has been defined, an iteration will be taken
+            for i, call_num in enumerate(call_nums):
+                if call_nums != i:
+                    slm_mask_dmm = i
+            # Have to wait for the SLM Mask to be configured to finish
+            # configuring the last Detuning Maps
+            for call_num, ser_det_map in ser_det_maps.items():
+                if slm_mask_dmm is not None and call_num >= slm_mask_dmm:
+                    break
+                det_map = _deserialize_det_map(ser_det_map)
+                seq.config_detuning_map(detuning_map=det_map, dmm_id=dmm_id)
```

Comment:
Well now I am thinking that everything would be solved by taking "config_detuning_map" as an operation instead of putting in "detuning_map". What do you think of this ? Eventually if we are attached to having the information regarding the configured detuning map in the schema I can have both.

Other idea: I pass the slm_mask_dmm, this provides the number of times the dmm used for the SLM can be configured here, and the last configurations are performed after the SLM is configured (all I am doing here is very much to identify what is the dmm_id used for the SLM and when should the configuration of the SLM happen compared to the configuration of this dmm_id)

Reply:
I think you're right, we should probably deal with it like a regular operation